templates/express/remix.config.js

```diff
@@ -1,6 +1,7 @@
 /** @type {import('@remix-run/dev').AppConfig} */
 module.exports = {
   ignoredRouteFiles: ["**/.*"],
+  devServerBroadcastDelay: 100,
```

Comment:
This PR introduces a race condition between the server reloading and the client assets finishing to build, which is why a server broadcast delay is required.

I think a better solution is for us to provide a mechanism for explicitly managing the Remix `watch` compile cycle in your dev server (whether Remix Dev Server or an external dev server).

I'm going to explore this a bit and get back to you.

Reply:
I think the race happens after building both server and client builds, and happens between notifying via websocket+reload on browser vs chokidar. In this case, it's not that bad as it takes constant time compared to app code size

Anyway, a mechanism for a custom server to get notified before livereload happens would be awesome!